src/libs/elektra/cmount.c

```diff
@@ -0,0 +1,2919 @@
+			return false;
+		}
+
+		strcpy (tmpStr, "user:");
+		strcat (tmpStr, mpName);
+		if (ksLookupByName (alreadyUsedMountpoints, tmpStr, KDB_O_NONE))
+		{
+			// TODO: set error key (was exception in c++ code)
+			fprintf (stderr, "Cascading mountpoint %s not possible, because user mountpoint already exists.\n", mpName);
+			ksDel (backendInfos);
+			ksDel (alreadyUsedMountpoints);
+			return false;
+		}
+
+		strcpy (tmpStr, "system:");
+		strcat (tmpStr, mpName);
+		if (ksLookupByName (alreadyUsedMountpoints, tmpStr, KDB_O_NONE))
+		{
+			// TODO: set error key (was exception in c++ code)
+			fprintf (stderr, "Cascading mountpoint %s not possible, because system mountpoint already exists.\n", mpName);
+			ksDel (backendInfos);
+			ksDel (alreadyUsedMountpoints);
+			return false;
+		}
+
+		elektraFree (tmpStr);
+	}
+	else
+	{
+		if (ksLookupByName (alreadyUsedMountpoints, mpName, KDB_O_NONE))
+		{
+			// TODO: set error key (was exception in c++ code)
+			// TODO: list used names in error message (see /src/libs/tools/src/backend.cpp:213)
+			fprintf (stderr, "Mountpoint %s is one of the already used names!\n", mpName);
+			ksDel (backendInfos);
+			ksDel (alreadyUsedMountpoints);
+			return false;
+		}
+	}
+
+	ksDel (alreadyUsedMountpoints);
+	ksDel (backendInfos);
+
+	/* TODO: STEP 4: Check if mounted below system:/elektra */
+	Key * elektraCheck = keyDup (mountPoint, KEY_CP_NAME);
+
+	/* Remove namespace */
+	for (int i = 0; i < keyGetNameSize (elektraCheck); ++i)
+	{
+		if (*(mpName + i) == ':')
+		{
+			keySetName (elektraCheck, mpName + i + 1);
+		}
+	}
+
+	Key * elektraKey = keyNew ("/elektra", KEY_END);
+	if (keyIsBelowOrSame (elektraKey, elektraCheck))
+	{
+		// TODO: set error key (was exception in c++ code)
+		fprintf (stderr, "Mountpoint %s is below the reserved names /elektra because it would cause inconsistencies in this or future versions.\n", mpName);
+	}
+
+	keyDel (elektraKey);
+	keyDel (elektraCheck);
+
+
+	/* Everything worked */
+	return true;
+}
+
+/**
+ * @brief give info about current mounted backends
+ *
+ * @param mountConf a keyset that contains everything below
+ * DEFAULT_MOUNTPOINTS_PATH
+ *
+ * @return A KeySet with Keys that contain information about mounted backends.
+ * The keyname is the mountpoint, the value of the key is the path.
+ * The returned KeySet has to be freed (ksDel()) by the caller.
+ */
+// C++: backends.cpp[28-50]
+KeySet * getBackendInfo (KeySet * mountConf)
+{
+	if (!mountConf || ksGetSize (mountConf) <= 0)
+	{
+		return NULL;
+	}
+
+	Key * rootKey = keyNew (DEFAULT_MOUNTPOINTS_PATH, KEY_END);
+	KeySet * result = ksNew (0, KS_END);
+
+	if (!rootKey || !result)
+	{
+		/* TODO: error handling */
+	}
+
+	for (elektraCursor it = 0; it < ksGetSize (mountConf); ++it)
+	{
+		Key * cur = ksAtCursor (mountConf, it);
+		if (keyIsDirectlyBelow(rootKey, cur))
+		{
+			size_t lenLookup = keyGetNameSize (cur); /* includes \0 */
+			if (lenLookup > 1)
+			{
+				lenLookup += strlen ("/definition/path");
```

Comment:
That key is specific to the `backend` plugin. We need a more general solution...